plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTaskImpl.java

```diff
@@ -34,7 +40,53 @@
 import com.diffplug.spotless.PaddedCell;
 
 @CacheableTask
-public class SpotlessTaskImpl extends SpotlessTask {
+public abstract class SpotlessTaskImpl extends SpotlessTask {
+	final File projectDir;
+
+	@Internal
+	public File getProjectDir() {
+		return projectDir;
+	}
+
+	Formatter buildFormatter() {
+		// <sketchy configuration cache trick>
+		SpotlessTaskImpl original = SpotlessTaskService.instance().get(getPath());
+		if (original == this) {
+			// a SpotlessTask is registered with the SpotlessTaskService **only** if configuration ran
+			// so if we're in this block, it means that we were configured
+			return Formatter.builder()
+					.lineEndingsPolicy(lineEndingsPolicy)
+					.encoding(Charset.forName(encoding))
+					.rootDir(getProjectDir().toPath())
+					.steps(steps)
+					.exceptionPolicy(exceptionPolicy)
+					.build();
+		} else {
+			// if we're in this block, it means that configuration did not run, and this
+			// task was deserialized from disk. All of our fields are ".equals" to their
+			// originals, but their transient fields are missing, so we can't actually run
+			// them. Luckily, we saved the task from the original, just so that we could restore
+			// its formatter, whose transient fields are fully populated.
+			return original.buildFormatter();
+		}
+		// </sketchy configuration cache trick>
+	}
```

Comment:
One answer to https://discuss.gradle.org/t/how-to-determine-if-configuration-cache-is-enabled/41383 (pt 1 of 2)